schematools/importer/base.py

```diff
@@ -478,19 +478,54 @@ def index_factory(
     In case of temporal data, this will lead to intersection tables a.k.a. through tables to
     accomodate n:m relations.
 
+    FK index:
+    In the JSON schema definition of the table, relations may be 1:N relations definied.
+    Where the child table is referencing a parent table.
+    These reference are not enforced by a database foreign key constraint. But will be
+    used in joins to collect data when calling a API endpoint. Therefore must be indexed
+    for optimal performance.
+
     The returned Index objects are keyed on the name of table
     """
 
     index = dict()
     metadata = metadata or MetaData()
     logger = LogfileLogger(logger) if logger else CliLogger()
+    table_object = metadata.tables[dataset_table.db_name()]
+    indexes_to_create = []
+
+    def make_hash_value(index_name):
+        """
+        Postgres DB holds currently 63 max charakters for objectnames.
+        To prevent exceeds and collisions, the index names are shortend
+        based upon a hash.
+        SHA1 holds a max output of 40 characters
+        """
+        hash = hashlib.sha1()
+        hash.update(bytes(index_name, "utf-8"))
+        index_name = hash.hexdigest() + "_idx"
+        return index_name
+
+    def define_fk_index():
+        """creates an index on the columns that refer to another table as
+        a child (foreign key reference)"""
+
+        if dataset_table.get_fk_fields():
+
+            for field in dataset_table.get_fk_fields():
+                field_name = f"{to_snake_case(field)}_id"
+                index_name = f"{db_table_name}_{field_name}_idx"
+                if len(index_name) > 64:
```

Comment:
You might want to use a constant for this value. Should this ever change in the default PostgreSQL installation or when basis_data custom compiles PostgreSQL we should be able to track this down easily. PostgreSQL calls this value `[NAMEDATALEN](https://www.postgresql.org/docs/11/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS)`. Maybe we should do:

```python
PSQL_NAMEDATALEN = 64
```

This also allows you to to add an `assert` statement to the function `make_hash_value`, like:

```python
assert len(index_name) <= (
    PSQL_NAMEDATALEN - 1
), f"index names should be shorter than {PSQL_NAMEDATALEN}"
```
This guards against future Python developers that mindlessly change the hash function, because they've read somewhere that `sha1` is "insecure", change it to `sha256` or something else and suddenly we end up with larger index name than PostgreSQL can handle.

Reply:
Goeie, er bestond al een constante voor object namen in Postgres. Die heb ik gebruikt. De assert heb ik gebruikt in een extra test op index naam.